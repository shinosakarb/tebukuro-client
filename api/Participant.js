// @flow
import Base from './Base'

export default class Participant extends Base {
}
