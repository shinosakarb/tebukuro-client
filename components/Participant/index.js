// @flow
import React, { Component } from 'react'

type Props = {
  id: number,
  eventId: number,
  name: string,
  onWaitingList: boolean,
  onCancel: Function,
}

export default class Participant extends Component<Props> {
  onCancelHandler = () => {
    this.props.onCancel({
      id: this.props.id,
      eventId: this.props.eventId,
    })
  }

  render() {
    const margin = { marginRight: 10 }
    return (
      <div>
        <span style={margin}>
          { this.props.name }
        </span>
        <span style={margin}>
          { this.props.onWaitingList ?
            'キャンセル待ち' : '参加可能'
          }
        </span>
        <button onClick={this.onCancelHandler} >
          キャンセル
        </button>
      </div>
    )
  }
}
