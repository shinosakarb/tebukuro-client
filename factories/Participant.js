export default {
  participant1: {
    id: 1,
    eventId: 1,
    name: 'participant1',
  },
  participant2: {
    id: 2,
    eventId: 1,
    name: 'participant2',
  },
}
