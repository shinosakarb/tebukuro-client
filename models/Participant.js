// @flow
import Record from './Base'

export const ParticipantRecord = Record({
  id: 0,
  name: '',
})

export default class Participant extends ParticipantRecord {
}
