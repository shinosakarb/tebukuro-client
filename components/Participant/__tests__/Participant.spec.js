import React from 'react'
import { mount, shallow } from 'enzyme'
import shallowToJson from 'enzyme-to-json'
import Participant from '../index'

const testProps = {
  id: 1,
  eventId: 2,
  name: 'Participant 1',
}

describe('Participant', () => {
  it('renders participant component', () => {
    const wrapper = shallow(<Participant {...testProps} onCancel={jest.fn()} />)
    const tree = shallowToJson(wrapper)

    expect(tree).toMatchSnapshot()
  })

  describe('when clicked', () => {
    const onCancel = jest.fn()
    const wrapper = mount(<Participant {...testProps} onCancel={onCancel} />)

    wrapper.find('button').simulate('click')

    it('should call onCancel once.', () => {
      expect(onCancel).toHaveBeenCalledTimes(1)
    })

    it('should call onCancel with correct argument.', () => {
      expect(onCancel).toBeCalledWith({
        id: testProps.id,
        eventId: testProps.eventId,
      })
    })
  })
})
