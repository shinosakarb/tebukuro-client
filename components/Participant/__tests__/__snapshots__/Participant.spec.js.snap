// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant with admitted participant renders participant component with admitted message. 1`] = `
<div>
  <span
    style={
      Object {
        "marginRight": 10,
      }
    }
  >
    Participant 1
  </span>
  <span
    style={
      Object {
        "marginRight": 10,
      }
    }
  >
    参加可能
  </span>
  <button
    onClick={[Function]}
  >
    キャンセル
  </button>
</div>
`;

exports[`Participant with waitlisted participant renders participant component with waitlisted message. 1`] = `
<div>
  <span
    style={
      Object {
        "marginRight": 10,
      }
    }
  >
    Participant 1
  </span>
  <span
    style={
      Object {
        "marginRight": 10,
      }
    }
  >
    キャンセル待ち
  </span>
  <button
    onClick={[Function]}
  >
    キャンセル
  </button>
</div>
`;
